File: Java/Open_MainActivity2/MainActivity.java
package com.example.java1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    private Button button2;

    TextView theTxt;
    Button btn_Button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

       /*
        TextView textView = (TextView) findViewById(R.id.textView1);
        textView.setText("This is fun");
        */

       theTxt = (TextView) findViewById(R.id.textView2);
       btn_Button = (Button) findViewById(R.id.button1);

       //Change the button text
       Button button_btn = (Button)findViewById(R.id.button1);
       button_btn.setText("Click me");

       //set the text to uppercase & or .toLowerCase (if there is)
       btn_Button.setOnClickListener(new View.OnClickListener(){
           @Override
           public void onClick(View v)
           {
               theTxt.setText("tanttax".toLowerCase());
           }
       });


       //Button to open another mainactivity2
        button2 = (Button) findViewById(R.id.button2);
        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openActivity2(); //Need methods
            }
        });

    } //TODO:: Oncreate ENDS HERE

    //TODO:: methodi edellisen button2 kanssa
    public void openActivity2() {
        Intent intent = new Intent (this, Main2Activity.class);
        startActivity(intent);
    }

}

File: string-array_Main2Activity/MainActivity.kt
package com.example.location2

import android.app.SearchManager
import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.widget.SearchView
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        textView.setText("Something happening here");
        button.setText("Main2Activity")

        //TODO:: Button click open to Main2activity
        val button = findViewById<Button>(R.id.button)
        button.setOnClickListener{
            val intent = Intent(this, Main2Activity::class.java)
            startActivity(intent)
        }
    }

    //TODO: Option menu search haku kohde
    override fun onCreateOptionsMenu(menu: Menu?): Boolean {

        val inflater = menuInflater
        inflater.inflate(R.menu.main_menu, menu) //TODO:: read folder --> res/menu/main_menu.xml

        val manager = getSystemService(Context.SEARCH_SERVICE) as SearchManager
        val searchItem = menu?.findItem(R.id.search)
        val searchView = searchItem?.actionView as SearchView

        searchView.setSearchableInfo(manager.getSearchableInfo(componentName))

        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String?): Boolean {
                searchView.clearFocus()
                searchView.setQuery("", false)
                searchItem.collapseActionView()
                Toast.makeText(this@MainActivity, "Looking for $query", Toast.LENGTH_LONG).show()
                return true
            }

            override fun onQueryTextChange(newText: String?): Boolean {
                Toast.makeText(this@MainActivity, "Looking for $newText", Toast.LENGTH_SHORT).show()

                return false
            }

        })
            return true

    } //TODO:: onCreateOptionsMenu ENDS HERE

}
